MSAL/src/MSALPublicClientApplication.m

```diff
@@ -244,73 +246,47 @@ - (id)initWithClientId:(NSString *)clientId
     
     _sliceParameters = [MSALPublicClientApplication defaultSliceParameters];
     
+    MSIDAADNetworkConfiguration.defaultConfiguration.aadApiVersion = @"v2.0";
     _expirationBuffer = 300;  //in seconds, ensures catching of clock differences between the server and the device
     
     return self;
 }
 
+#pragma mark - Accounts
 
-- (NSArray <MSALAccount *> *)accounts:(NSError * __autoreleasing *)error
+- (NSArray <MSALAccount *> *)allAccounts:(NSError * __autoreleasing *)error
 {
-    NSError *msidError = nil;
-    __auto_type host = self.authority.msidAuthority.environment;
-    __auto_type msidAccounts = [self.tokenCache allAccountsForEnvironment:host
-                                                                 clientId:self.clientId
-                                                                 familyId:nil
-                                                                  context:nil
-                                                                    error:&msidError];
-
-    if (msidError)
-    {
-        *error = msidError;
-        return nil;
-    }
-
-    NSMutableSet *msalAccounts = [NSMutableSet new];
-
-    for (MSIDAccount *msidAccount in msidAccounts)
-    {
-        MSALAccount *msalAccount = [[MSALAccount alloc] initWithMSIDAccount:msidAccount];
-
-        if (msalAccount)
-        {
-            [msalAccounts addObject:msalAccount];
-        }
-    }
-
-    return [msalAccounts allObjects];
+    MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
+                                                                         authority:nil
+                                                                          clientId:self.clientId];
+    return [request accounts:error];
 }
 
 - (MSALAccount *)accountForHomeAccountId:(NSString *)homeAccountId
                                    error:(NSError * __autoreleasing *)error
 {
-    NSArray<MSALAccount *> *accounts = [self accounts:error];
-
-    for (MSALAccount *account in accounts)
-    {
-        if ([account.homeAccountId.identifier isEqualToString:homeAccountId])
-        {
-            return account;
-        }
-    }
-
-    return nil;
+    MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
+                                                                         authority:self.authority
```

Comment:
This doesn't need to filter by authority anymore, right?

Reply:
done